docs/design/localization.rst

```diff
@@ -0,0 +1,217 @@
+
+    from radical.pilot.utils.misc import get_resource_fs_url
+    import radical.saga as rs
+    rs.filesystem.Directory(get_resource_fs_url(...)).get_cwd()
+
+We would want to be sure to get a consistent *resource* and *access* value from
+the :ref:`scalems.radical parser <scalems.radical command line>` or `scalems.radical.runtime.Configuration`.
+
+.. seealso::
+
+    * `radical.pilot.utils.misc.get_resource_fs_url`
+    * https://github.com/radical-cybertools/radical.pilot/issues/2668#issuecomment-1236184810
+    * https://github.com/radical-cybertools/radical.pilot/issues/2668#issuecomment-1239956265
+
+Get a portable URI for an RP URI
+--------------------------------
+
+This is only possible, generally, if the (``task:///``, ``pilot:///``, *etc*)
+RP URI is in a subdirectory of the filesystem endpoint URI.
+RP uses `radical.saga.filesystem.Directory` to navigate from the filesystem endpoint
+to the Session working directory, which may not be representable in a derived URI.
+
+Determine the actual filesystem paths of Tasks
+----------------------------------------------
+
+The Task working directory is the :py:attr:`radical.pilot.Task.sandbox`.
+``task:///`` URIs and relative paths can be interpreted with respect to
+the sandbox path, which is determined during task submission on the client,
+and which exists by the time the Task reaches **some state**.
+
+.. warning:: Sandbox representation may be wrong for raptor tasks: https://github.com/radical-cybertools/radical.pilot/issues/2802
+
+from a Task
+^^^^^^^^^^^
+
+.. code-block:: python
+
+    this_task_path = os.getcwd()
+
+Anything else would need to be provided by the client or raptor Master task,
+or inferred in a hacky way by inspecting parent directories.
+
+In particular, there is no way to know which RP Resource the Pilot is running under.
+
+In general, Tasks do not have access to the RP Session under which they are running,
+so they know less about other Tasks than the client
+(with the exception of raptor Master tasks).
+
+from the client
+^^^^^^^^^^^^^^^
+
+.. admonition:: **Question**
+
+    Are sandbox URIs reliably constructed from, say, the ``filesystem_endpoint`` URL?
+    I.e. can we anticipate the ``Url.scheme`` and expect it to stay consistent?
+
+.. code-block:: python
+
+    task_dir_url = radical.utils.Url(task.sandbox)
+    if task_dir_url.scheme == 'file':
+        # Find the absolute path.
+        task_dir = task_dir_url.path
+        # Find a reproducible way to regain access to the path.
+        resource = ...
+        ...
+    elif task_dir_url.scheme.endswith('sftp'):
+        # Find the absolute path.
+        ...
+        # Find a reproducible way to regain access to the path.
+        ...
+
+.. code-block:: python
+
+    radical.saga.filesystem.File(task_dir_url, session=session)
+    ...
+
+.. warning::
+
+    RP casts absolute paths in staging directives to ``file:///`` URIs,
+    which are always considered in the client filesystem. This is different
```

Comment:
Absolute paths can also refer to the remote resource's file system, for example if used for an input staging target.

Reply:
That does not seem to be true, in RP 1.21, at least. Ref: https://github.com/radical-cybertools/radical.pilot/issues/2848